### PR TITLE
Fix last name sorting 

### LINKED ISSUE #373: if 'last name' sorting has been set the comma between the last and first name should be suppressed when no first name has been entered
## Expected behavior
If the sorting order `last name` has been set, the comma which usually separates the last and first name, should be suppressed when no first name has been entered.

## Current behavior
A comma behind the last name is always shown independently if a first name has been entered or not.

20171003_contacts_commata

### Steps to reproduce
1. Open the contacts app
2. Set the sorting order to 'last name'
3. check-out a contact for which no first name has been entered.

#### Server Configuration
OS: Linux 3.16.47
Web server: Apache2 2.4.26
Database: MariaDB 10.0.32
PHP version: 5.6.30
Nextcloud version: 12.03
Contacts app version: 2.0.0

#### Client Configuration
Browser: Mozilla Firefox 56.0
Operating system: Windows 7

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/50016534-if-last-name-sorting-has-been-set-the-comma-between-the-last-and-first-name-should-be-suppressed-when-no-first-name-has-been-entered?utm_campaign=plugin&utm_content=tracker%2F46751899&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F46751899&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by singhalpiyush12**
Fix #373

### CODE REVIEW

File: js/components/contact/contact_controller.js [hunk 1]
```diff
@@ -18,13 +18,19 @@ angular.module('contactsApp')
 			return ctrl.contact.displayName();
 		}
 
-		if (SortByService.getSortBy() === 'sortLastName') {
+		if (SortByService.getSortBy() === 'sortLastName'&ctrl.contact.firstName() !=='') {
```
Comment (skjnldsv) [inline js/components/contact/contact_controller.js]: please add spaces between `&` and `!==` :)

File: js/components/contact/contact_controller.js [hunk 2]
```diff
@@ -18,13 +18,19 @@ angular.module('contactsApp')
...
 			return (
 				ctrl.contact.lastName() + ', '
 				+ ctrl.contact.firstName() + ' '
 				+ ctrl.contact.additionalNames()
 			).trim();
 		}
+		else if (SortByService.getSortBy() === 'sortLastName'&ctrl.contact.firstName() ==='') {
```
Comment (skjnldsv) [inline js/components/contact/contact_controller.js]: Same here! ;)

**Comment by mburnicki** [comment]
There are also cases where the first name is empty, e.g. if I have "Alberto Pizzeria" or "Hilton Hotel" then the full name is saved in the field for last name, and there is no entry for the first name.

BTW, in 2.0.1, if I change the sort order, "something" changes, but not as expected. Not yet sure if this is due to the contacts coming from quite a number of different address books.

Anyway, in the contacts app available for owncloud you could select "last name, first name" as sort order, and if you did then the contacts list on the left showed e.g.

Alberto Pizzeria
Burnicki, Martin
Hilton Hotel
... etc.

Now the list still shows the first name first, and then the last name, even if sort order has been changed to "last name". This is pretty confusing if you have many contacts (I've got ~2800 in ~40 address books).

And even though the first name / last name / display name are set properly if you look at the details pages, the contacts list on the left is still sorted by first name, even if I have selected "last name" in the settings.

BTW, I'm using German language, is it possible that the sort order selection field is not evaluated properly if a non-English languarge is used for the GUI?

**Comment by skjnldsv** [comment]
Hey! Any news? :)

**Comment by skjnldsv** [comment]
Yes

**Pull request closed without merging**